src/components/plugins/Plugin.tsx

```diff
@@ -0,0 +1,42 @@
+import {
+  IPluginState as EntityData,
+  IProposalState as ProposalData,
+  IProposalBaseCreateOptions as ProposalOptions,
+  AnyPlugin,
+  ProposalPlugin as ProposableEntity,
+} from "@dorgtech/arc.js";
+import { Observable } from "rxjs";
+
+export interface IPlugin<TPluginEntity extends AnyPlugin> {
+  state(): Observable<EntityData>;
+  fetchState(): Promise<EntityData>;
+}
+
+type InferPlugin<TPlugin> = TPlugin extends IPlugin<infer TPluginEntity>
+  ? TPluginEntity
+  : undefined;
+
+export class Plugin<
```

Comment:
This is going to be the main class of a plugin, I was thinking that we can create two interfaces, which are going to be the `UnknownPlugin` and the `ProposalPlugin` - This way, whenever we want to create a new plugin instance we would just need to implement one of those interfaces when initialising and have access to every method of the plugin